ingestion/src/metadata/ingestion/source/database/bigtable/metadata.py

```diff
@@ -0,0 +1,192 @@
+        connection: BigTableConnection = config.serviceConnection.__root__.config
+        if not isinstance(connection, BigTableConnection):
+            raise InvalidSourceException(
+                f"Expected BigTableConnection, but got {connection}"
+            )
+        return cls(config, metadata)
+
+    def get_configured_database(self) -> Optional[str]:
+        return None
+
+    def get_database_names(self) -> Iterable[str]:
+        return self.get_database_names_raw()
+
+    def get_database_names_raw(self) -> Iterable[str]:
+        yield from self.client.project_ids()
+
+    def get_schema_name_list(self) -> List[str]:
+        project_id = self.context.database
+        try:
+            # the first element is a list of instances
+            # the second element is another collection (seems empty) and I do not know what is its purpose
+            instances, _ = self.client.list_instances(project_id=project_id)
+            self.instances[project_id] = tuple(instances)
+        except Exception as err:
+            logger.debug(traceback.format_exc())
+            logger.error(
+                f"Failed to list BigTable instances in project {project_id}: {err}"
+            )
+            raise
+        return [instance.instance_id for instance in self.instances[project_id]]
+
+    def get_table_name_list(self, schema_name: str) -> List[str]:
+        project_id = self.context.database
+        try:
+            instance = [
+                i for i in self.instances[project_id] if i.instance_id == schema_name
+            ][0]
+            tables = instance.list_tables()
+            self._set_nested(self.tables, [project_id, instance.instance_id], tables)
+            return [t.table_id for t in self.tables[project_id][instance.instance_id]]
+        except Exception as err:
+            logger.debug(traceback.format_exc())
+            # add context to the error message
+            logger.error(
+                f"Failed to list BigTable table names in {project_id}.{schema_name}: {err}"
+            )
+        return []
+
+    def get_table_constraints(
+        self, db_name: str, schema_name: str, table_name: str
+    ) -> List[TableConstraint]:
+        return [
+            TableConstraint(
+                constraintType=ConstraintType.PRIMARY_KEY, columns=["row_key"]
+            )
+        ]
+
+    def get_table_columns_dict(
+        self, schema_name: str, table_name: str
+    ) -> Union[List[Dict], Dict]:
+        project_id = self.context.database
+        try:
+            table = self._find_table(project_id, schema_name, table_name)
+            if table is None:
+                raise RuntimeError(
+                    f"Table {project_id}/{schema_name}/{table_name} not found."
+                )
+            column_families = table.list_column_families()
+            records = [{"row_key": b"row_key"}]
+            for cf in list(column_families.keys())[:MAX_COLUMN_FAMILIES]:
```

Comment:
can you add a comment as to why we fetch up to `MAX_COLUMN_FAMILIES`. This will help us understand better in the future.

Reply:
addressed in
- https://github.com/open-metadata/OpenMetadata/pull/15122/files#diff-f40b868710d9c5f65dd0b0ad49b889bb5da564e0b6b9b4dd5f1dcec2a547d29fR45-R48
- https://github.com/open-metadata/OpenMetadata/pull/15122/files#diff-f40b868710d9c5f65dd0b0ad49b889bb5da564e0b6b9b4dd5f1dcec2a547d29fR160-R161